smartlearner/trainer.py

```diff
@@ -42,6 +45,26 @@ def build_theano_graph(self):
                                       name="learn")
         #theano.printing.pydotprint(self._learn, '{0}_learn_{1}'.format(self._optimizer.loss.model.__class__.__name__, theano.config.device), with_ids=True)
 
+    def save(self, path):
+        savedir = utils.create_folder(pjoin(path, "training"))
+        self.status.save(savedir)
+        self._optimizer.save(savedir)
+        self._batch_scheduler.save(savedir)
+
```

Comment:
The trainer doesn't have an reference to the model. The model is save via the optimizer.